middlewares/authenticator.go

```diff
@@ -88,6 +105,23 @@ func parserDigestUsers(users types.Users) (map[string]string, error) {
 	return userMap, nil
 }
 
+func getLinesFromFile(filename string) ([]string, error) {
+	dat, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	// Trim lines and filter out blanks
+	rawLines := strings.Split(string(dat), "\n")
+	filteredLines := rawLines[:0]
```

Comment:
I read somewhere that I could bypass an allocation this way. It's allegedly marginally faster. If you want to take that small hit for the sake of clarity, I am not opposed. Just let me know.